tests/upgrade_path/test_upgrade_path.py

```diff
@@ -0,0 +1,198 @@
+import pytest
+import logging
+import json
+import os
+from common import reboot
+from jinja2 import Template
+import ipaddr as ipaddress
+from ptf_runner import ptf_runner
+from common.platform.ssh_utils import prepare_testbed_ssh_keys
+from datetime import datetime
+
+from common.fixtures.ptfhost_utils import copy_ptftests_directory   # lgtm[py/unused-import]
+from common.fixtures.ptfhost_utils import change_mac_addresses      # lgtm[py/unused-import]
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+logger = logging.getLogger(__name__)
+
+SOURCE_UPGRADE_SCRIPT = 'upgrade_path/upgrade_sonic.sh'
+UPGRADE_SCRIPT = '/tmp/upgrade_sonic.sh'
+
+TMP_VLAN_PORTCHANNEL_FILE = '/tmp/portchannel_interfaces.json'
+TMP_VLAN_FILE = '/tmp/vlan_interfaces.json'
+TMP_PORTS_FILE = '/tmp/ports.json'
+
+
+@pytest.fixture(scope="module")
+def duthost_original_image(duthost):
+    original_image = ""
+    res = duthost.shell('sonic_installer list', module_ignore_errors=True)
+    for line in res['stdout'].splitlines():
+        if line.startswith('Current:'):
+            original_image = line.split(':')[-1].strip()
+            break
+    logger.info("Original image is {}".format(original_image))
+    return original_image
```

Comment:
You can use the `image_facts` module for this purpose:
https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/image_facts.py

Reply:
There is duthost method here too: https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices.py#L502

Reply:
The duthost_original_image method is removed and a new parameter is added to perform restore job.